enterprise/views.py

```diff
@@ -294,3 +297,59 @@ def post(self, request):
             # We're handing consent only for a particular course
             return self.post_course_specific_consent(request, specific_course, consent_provided)
         return self.post_account_consent(request, consent_provided)
+
+
+class RouteLearnerData(APIView):
+    """
+    Provide an endpoint to route learner data to the appropriate channels based on Enterprise's configuration
+    """
+
+    def get(self, request, format=None):
+        """
+        It is unclear if a get request should do anything at this point.
+        """
+        return Response()
+
+    def post(self, request, format=None):
+        """
+        This endpoint is meant to be hit when certificates are generated or when audit learners in self paced courses
+        reach a passing grade. When this happens, this method will route the learners data appropriately.
+
+        For a given learner who has completed a course, find out the following:
+            if the learner is associated with an enterprise customer.
+            if the course is part of an enterprise enrollment.
+            if the enterprise customer is configured for SuccessFactors.
+
+        Continue if all the conditions above are met, otherwise abort processing.
+
+        Based on the EnterpriseCustomerPluginConfiguration for the Enterprise, determine which endpoints are
+        configured to receive learner data, and send a formatted request to each of those endpoints.
```

Comment:
@brittneyexline I believe the following docs should provide you with a good starting point:

* http://docs.celeryproject.org/en/latest/userguide/tasks.html
* http://docs.celeryproject.org/en/latest/userguide/calling.html

In short, the idea is to have edx-enterprise provide a celery task that would notify integration endpoints. The task would be called asynchronously (with `task.apply_async`). But please make sure task result (and, especially, any errors happened) are not just discarded, but at least logged. Better yet to have it exposed to admin - as a DB model + admin integration, but that's slightly more involving, so it would be nice to have, but log record would be sufficient.